### PR TITLE
add a CI "platform" with `assert()`s enabled (`NDEBUG` not defined)

### PR DESCRIPTION
**Pull request by spoonincode**
A demo run with an `assert()` failure is at
https://github.com/AntelopeIO/leap/actions/runs/8150929931

### CODE REVIEW

**Review by heifner — APPROVED** [review]

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -135,6 +135,7 @@ jobs:
         include:
           - cfg: {name: 'ubuntu20',      base: 'ubuntu20', builddir: 'ubuntu20'}
           - cfg: {name: 'ubuntu22',      base: 'ubuntu22', builddir: 'ubuntu22'}
+          - cfg: {name: 'asserton',      base: 'asserton', builddir: 'asserton'}
```
Comment (linh2931) [inline .github/workflows/build.yaml:138]: why `asserton`, not `assertion`? Name length restriction?

**Review by linh2931 — APPROVED** [review]

File: .github/workflows/build.yaml (hunk 1 above)
Comment (heifner) [reply .github/workflows/build.yaml:138]: FWIW, I read it as `assert on`

Comment (linh2931) [reply .github/workflows/build.yaml:138]: Haha, I read it as `asser` `ton`.

**Comment by ericpassmore** [comment]
Note:start
group: STABILITY
category: TEST
summary: Adds a test pass with `assert()`s enabled.
Note:end

**Review by greg7mdp — APPROVED** [review]

**Pull request merged**